subsys/shell/Kconfig.backends

```diff
@@ -167,6 +167,92 @@ source "subsys/logging/Kconfig.template.log_config"
 
 endif #SHELL_BACKEND_RTT
 
+config SHELL_BACKEND_TELNET
+	bool "Enable TELNET backend."
+	help
+	  Enable TELNET backend.
+
+if SHELL_BACKEND_TELNET
+
+config SHELL_BACKEND_TELNET_PORT
```

Comment:
imo, you can change prefix to `SHELL_TELNET_<..>`.  It's rather clear that it's a backend. I know that serial has backend in the name but rtt does not so there is no consistency and i would prefer to remove `_BACKEND_` part from other backend in the future to make those defines shorter.